### ISSUE TITLE
Wrong error message when device is offline

### ISSUE THREAD

**Issue by pmellaaho** [aws-amplify/amplify-ui-android#151]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui-android/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui-android/blob/main/CONTRIBUTING.md#reporting-bugsfeature-requests).

### Which UI component?

Authenticator

### Gradle script dependencies

```groovy
// Put output below this line
authenticatorVersion = "1.1.0"
```

### Environment information

```
# Put output below this line
Gradle 8.6

```

### Describe the bug

The Authenticator is showing wrong error message's when e.g. device is offline and user tries to Sign-in. In can see that there is a valid text ("Please check your connectivity") defined for these cases but it just isn't used. It seems that AuthenticatorViewModel is mapping UnknownHostException -cause to the this error message but for some reason the mapping code doesn't work and ambigious "Sorry, something went wrong" -message is shown to the user. With AndroidLoggingPlugin enabled I can see the following lines in Logcat:

`Auth State Change: Configured(authNState=SigningIn(signInState=SigningInWithSRP(srpSignInState=Error(exception=aws.smithy.kotlin.runtime.http.HttpException: java.net.UnknownHostException: Unable to resolve host "cognito-idp.eu-central-1.amazonaws.com": No address associated with hostname))), authZState=SigningIn(id=))
Auth State Change: Configured(authNState=SignedOut(signedOutData=SignedOutData(lastKnownUsername=null, hostedUIErrorData=null, globalSignOutErrorData=null, revokeTokenErrorData=null)), authZState=Configured(id=))`

### Code Snippet

```java
// Put your code below this line.

```

### Log output

```
// Put your logs below this line
Auth State Change: Configured(authNState=SigningIn(signInState=SigningInWithSRP(srpSignInState=Error(exception=aws.smithy.kotlin.runtime.http.HttpException: java.net.UnknownHostException: Unable to resolve host "cognito-idp.eu-central-1.amazonaws.com": No address associated with hostname))), authZState=SigningIn(id=))
Auth State Change: Configured(authNState=SignedOut(signedOutData=SignedOutData(lastKnownUsername=null, hostedUIErrorData=null, globalSignOutErrorData=null, revokeTokenErrorData=null)), authZState=Configured(id=))

```

**Comment by mattcreaser** [comment]
Looks like Authenticator is looking for a direct cause of `UnknownHostException`, but there's an extra layer of `HttpException` wrapping the `UnknownHostException`. Authenticator should search through the causes instead of only looking one level deep.

**Comment by mattcreaser** [comment]
This has been resolved in Authenticator 1.2.1
